### PR TITLE
feat: add getJson and outputJson for plugin io util

### PR DESCRIPTION
**Pull request by nilslice**
Looks like `zig@master` changed some build APIs, so i'll figure that out and update this PR (and likely the SDK repo too)

### CODE REVIEW

File: src/main.zig [hunk 1]
```diff
@@ -34,6 +49,15 @@ pub const Plugin = struct {
         return buf;
     }
 
+    // IMPORTANT: It's the caller's responsibility to free the returned struct
+    pub fn getJson(self: Plugin, comptime T: type, options: std.json.ParseOptions) !Json(T) {
+        const bytes = try self.getInput();
+        const out = try std.json.parseFromSlice(T, self.allocator, bytes, options);
+        const FromJson = Json(T);
+        const input = FromJson{ .parsed = out, .slice = bytes };
```
Comment (nilslice) [inline src/main.zig:57]: I don't know of a way to better handle this such that the end-user can get the automatic type from the JSON, and have enough control over freeing the memory so it doesn't deallocate before a value is used. 

In the example here: 
```zig
const Input = struct {
    name: []const u8,
    age: u16,
};

export fn json_input() i32 {
    const plugin = Plugin.init(allocator);
    const json = plugin.getJson(Input, .{}) catch unreachable; // you must call .deinit() when done
    defer json.deinit();

    const input: Input = json.value();
    const out = std.fmt.allocPrint(allocator, "Hello, {s}. You are {d} years old!\n", .{ input.name, input.age }) catch unreachable;

    plugin.output(out);
    return 0;
}
```

The `input.name` (a `[]const u8`) would be freed unless `json.deinit()` is called after this function returns and the value is used outside this function. Initially I thought I could handle this for the end-user, but I don't _think_ I can. 

So, this returns the `Json` wrapped type, with methods `.value()` (to get the parsed struct), and `.deinit()` to give the end-user the ability to free the slice used to parse the struct.

Comment (zshipko) [reply src/main.zig:57]: I think you can avoid this by using `alloc_always` when parsing the JSON:

```
std.json.parseFromSlice(Config, allocator, data, .{.allocate = .alloc_always})
```

Comment (nilslice) [reply src/main.zig:57]: @zshipko thanks, this works great!

Comment (nilslice) [reply src/main.zig:57]: ok, now there is a `getJson` which returns the `comptime T` and handles the memory management. kind of the "expected / easy-mode" path. 

and, there's a `getJsonOpt` which is the previous implementation and allows the end user to control the parsing and allocation.

**Review by zshipko — APPROVED** [review]
Looks good!

**Comment by nilslice** [comment]
Going to update the README to match these changes (and today's updates to Zig :laughing:) and will merge.

**Pull request merged**